src/Microsoft.DotNet.Interactive.AIUtilities.Tests/FunctionTests.cs

```diff
@@ -0,0 +1,221 @@
+                                  },
+                                  "required": [
+                                    "a",
+                                    "b",
+                                    "c"
+                                  ]
+                                }
+                                """);
+
+
+    }
+
+    [Fact]
+    public void can_create_function_from_delegate_with_complex_type_as_return()
+    {
+        var declaration = GptFunction.Create((int a, string b, string[] c) => new Uri($"{a}.{b}.{c}"), "DoCompute");
+
+        declaration.JsonSignature.FormatJson().Should().Be("""
+                                                           {
+                                                             "name": "DoCompute",
+                                                             "parameters": {
+                                                               "type": "object",
+                                                               "properties": {
+                                                                 "a": {
+                                                                   "type": "integer"
+                                                                 },
+                                                                 "b": {
+                                                                   "type": "string"
+                                                                 },
+                                                                 "c": {
+                                                                   "type": "array",
+                                                                   "items": {
+                                                                     "type": "string"
+                                                                   }
+                                                                 }
+                                                               }
+                                                             },
+                                                             "results": {
+                                                               "type": "object"
+                                                             },
+                                                             "required": [
+                                                               "a",
+                                                               "b",
+                                                               "c"
+                                                             ]
+                                                           }
+                                                           """);
+    }
+
+    [Fact]
+    public void can_create_function_from_delegate_with_enums_as_parameters()
+    {
+        var declaration = GptFunction.Create((int a, double b, EnumType c) => $"{a} {b} {c}", "DoCompute");
```

Comment:
I would either put the name property first or give it a default value. Why does the caller care about the function name?

Reply:
done